pennylane/workflow/qnode.py

```diff
@@ -632,6 +634,22 @@ def add_transform(self, transform_container: TransformContainer):
         """
         self._transform_program.push_back(transform_container=transform_container)
 
+    def update(self, **kwargs) -> "QNode":
+        """Return a new QNode instance with updated constructor arguments."""
+        if not kwargs:
+            valid_params = (
+                set(self.init_args.copy().pop("gradient_kwargs"))
+                | qml.gradients.SUPPORTED_GRADIENT_KWARGS
+            )
+            raise ValueError(
+                f"Must specify at least one configuration property to update. Valid properties are: {valid_params}."
+            )
+        original_init_args = self.init_args.copy()
+        gradient_kwargs = original_init_args.pop("gradient_kwargs")
+        original_init_args.update(gradient_kwargs)
+        original_init_args.update(kwargs)
+        return type(self)(**original_init_args)

```

Comment:
This is an optional suggestion as it is completely harmless

```suggestion
        return QNode(**original_init_args)

```